### PR TITLE
Fix incorrect json unmarshalling during execution

### PR DESCRIPTION
**Pull request by nmaquet**
This PR fixes a bug where json.Unmarshal was failing due to an incorrect assumption and the error wasn't checked.

### CODE REVIEW

File: execution.go [hunk 1]
```diff
@@ -970,47 +973,32 @@ type insertionTarget struct {
...
 		}
-		return in
+		return parsedInput
 	case nil:
 		return nil
 	default:
-		panic(fmt.Sprintf("unhandled type: %s", reflect.TypeOf(in).Name()))
+		log.Panicf("unhandled type: %T", parsedInput)
 	}
+	return nil // unreachable
```
Comment (pkqk) [inline execution.go:1001]: Might as well pull the `default` case out of the switch statement and put it's panic here, then there isn't an unreachable branch.

Comment (nmaquet) [reply execution.go:1001]: You'll still have the same issue I think. The go compiler doesn't recognise `log.Panicf` as a panic.

Comment (pkqk) [reply execution.go:1001]: oh, I wasn't considering the compilers feelings, just that then you don't need to say that part is unreachable

**Comment by pkqk** [comment]
Closing this as the new execution work will replace it

**Pull request closed without merging**